lib/file_validators/validators/file_content_type_validator.rb

```diff
@@ -46,35 +49,13 @@ def parse_values(value)
         Array.wrap(value).reject(&:blank?)
       end
 
-      def get_file_path(value)
-        if value.try(:path)
-          value.path
-        else
-          raise ArgumentError, 'value must return a file path in order to validate file content type'
-        end
-      end
-
-      def get_file_name(value)
-        if value.try(:original_filename)
-          value.original_filename
-        else
-          File.basename(get_file_path(value))
+      def get_content_type(value, tool)
+        if tool.present?
+          return FileValidators::MimeTypeAnalyzer.new(tool).call(value)
         end
-      end
 
-      def get_content_type(value, mode)
```

Comment:
Sorry to be somewhat necroposting, but this needs some context.

I believe this is the change that broke CarrierWave support in our project.

`value` here is the actual field value from the model, supplied from the outside. The old version used to demand `#path` and optional `#original_filename` from the `value`, whereas the new edition requires `value` to be an `IO` instance.

Even though API breakages are expected due to major version bump, I would kindly suggest removing CarrierWave (others?) support and been tested notice from the README.

Seekers may also benefit from version `3.0.0.beta1` which was the last one that works in my case.

The actual error message when RSpec's failed for us was:

```
NoMethodError:
  undefined method `eof?' for #<VaultDocUploader:0x00007feaa4a46168 @model=#<VaultDoc id: nil ...>, @mounted_as=:document, @staged=true, @file=#<CarrierWave::SanitizedFile:0x000000000d962d20 @file="/app/tmp/vault-cache/1691776062-272660352770104-0001-2145/logo.png", @original_filename="logo.png", @content_type="image/png", @content=nil>, @filename="logo.png", @cache_id="1691776062-272660352770104-0001-2145", @identifier=nil, @versions={:thumb=>#<VaultDocUploader::Uploader22380:0x000000000d960750 @model=#<VaultDoc id: nil, document: nil, state: "uploaded", title: "My Document", original_filename: nil, thumb_present: true, created_at: nil, deleted_at: nil, vault_keeper_id: "93419cd4-87ca-42ec-8ae4-ecfda99b7620">, @mounted_as=:document, @staged=true, @file=#<CarrierWave::SanitizedFile:0x000000000d96a8e0 @file="/app/tmp/vault-cache/1691776062-272660352770104-0001-2145/thumb_logo.png", @original_filename="logo.png", @content_type="image/png", @content=nil>, @filename="logo.png", @cache_id="1691776062-272660352770104-0001-2145", @identifier=nil, @versions={}, @format=nil, @parent_version=#<VaultDocUploader:0x00007feaa4a46168 ...>, @parent_cache_id="1691776062-272660352770104-0001-2145", @original_filename="logo.png", @geometry={:width=>315, :height=>315}, @cache_storage=#<CarrierWave::Storage::File:0x000000000a47d088 @uploader=#<VaultDocUploader::Uploader22380:0x000000000d960750 ...>, @cache_called=nil>>}, @format=nil, @original_filename="logo.png", @cache_storage=#<CarrierWave::Storage::File:0x000000000d961628 @uploader=#<VaultDocUploader:0x00007feaa4a46168 ...>, @cache_called=nil>>
# /gems/file_validators-3.0.0/lib/file_validators/mime_type_analyzer.rb:27:in `extract_with_file'
# /gems/file_validators-3.0.0/lib/file_validators/mime_type_analyzer.rb:16:in `call'
# /gems/file_validators-3.0.0/lib/file_validators/validators/file_content_type_validator.rb:60:in `get_content_type'
```
... with `VaultDocUploader` being inherited from `CarrierWave::Uploader::Base`.